### PR TITLE
Report specification model

### PR DESCRIPTION
**Pull request by natechols**

### CODE REVIEW

**Comment by mpkocher** [comment]
Can you bump the pbcommand version?

File: tests/data/report-specs/report_spec.json [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            "id": "table1",
+            "title": "Table 1",
+            "description": "The first table",
+            "columns": [
+                {
+                    "header": "Column 1",
+                    "type": "int",
+                    "id": "column1",
+                    "description": "A column of type int",
+                    "format": "%d"
```
Comment (mpkocher) [inline tests/data/report-specs/report_spec.json]: Should this be converted to the new format?

Comment (natechols) [reply tests/data/report-specs/report_spec.json]: Yeah, I missed that - I really need to have a function that validates the formats too.

File: pbcommand/models/report.py [hunk 2]
```diff
@@ -838,3 +838,252 @@ def _merge_tables(tables):
...
+    def get_attribute_spec(self, id_):
+        return self._attr_dict.get(id_, None)
+
+    def get_plotgroup_spec(self, id_):
+        return self._plotgrp_dict.get(id_, None)
+
+    def get_table_spec(self, id_):
+        return self._table_dict.get(id_, None)
+
+    def validate_report(self, rpt, raise_if_found=True):
```
Comment (mpkocher) [inline pbcommand/models/report.py]: A suggestion for minor change. Can we have method `is_report_valid(rpt)` (or similar) return boolean and have `validate_report(rpt)` return rpt or raise if it's not valid?

Comment (natechols) [reply pbcommand/models/report.py]: The reason `validate_report` doesn't return the report is that I want to be able to test for errors in response to specific defects.

Comment (natechols) [reply pbcommand/models/report.py]: I guess I can just do some string processing of the exception messages, though.

Comment (mpkocher) [reply pbcommand/models/report.py]: Is the expected behavior of `validate_report(self, rpt, raise_if_found=False)` that the caller still needs to wrap in a try?

**Pull request merged**